pkg/syncer/storagepool/intended_state.go

```diff
@@ -189,34 +189,37 @@ func newIntendedState(ctx context.Context, ds *cnsvsphere.DatastoreInfo,
 	}, nil
 }
 
-func isRemoteVsan(ctx context.Context, dsprops *dsProps, clusterID string, vcclient *vim25.Client) (bool, error) {
+func isRemoteVsan(ctx context.Context, dsprops *dsProps, clusterIDs []string, vcclient *vim25.Client) (bool, error) {
 	log := logger.GetLogger(ctx)
 	// If datastore type is not vsan, then return false.
 	if dsprops.dsType != vsanDsType {
 		return false, nil
 	}
-	// Get vsan cluster uuid.
-	clsMoRef := types.ManagedObjectReference{
-		Type:  "ClusterComputeResource",
-		Value: clusterID,
-	}
-	cls := object.NewClusterComputeResource(vcclient, clsMoRef)
-	var clsMo mo.ClusterComputeResource
-	err := cls.Properties(ctx, clsMoRef, []string{"configurationEx"}, &clsMo)
-	if err != nil {
-		log.Errorf("Error fetching cluster properties for %s. Err: %+v", clusterID, err)
-		return false, err
-	}
-	vsanClsUUID := clsMo.ConfigurationEx.(*types.ClusterConfigInfoEx).VsanConfigInfo.DefaultConfig.Uuid
-	vsanClsUUID = strings.ReplaceAll(vsanClsUUID, "-", "")
-	dsContainerID := strings.ReplaceAll(dsprops.containerID, "-", "")
-	log.Debugf("Verifying whether vSAN Datastore %s with containerID: %s is local to cluster uuid %s",
-		dsprops.dsName, dsContainerID, vsanClsUUID)
-	// The vsan datastore is a remote mounted one if its containerID is not the
-	// same as vsan cluster uuid.
-	if dsContainerID != vsanClsUUID {
-		log.Infof("vSAN Datastore %s is remote to this cluster", dsprops.dsName)
-		return true, nil
+
+	for _, clusterID := range clusterIDs {
```

Comment:
We should only check the cluster where this datastore is present.
Otherwise we will end up saying that the given datastore is remotely mounted.

One way to do that is to maintain a cache between datastoreID and clusterID (refer to design spec for more details).

Reply:
I agree but another scenario comes to my mind looking at the way we have structured the `intendedState` - 
Is it possible for a datastore in one cluster to be mounted on another cluster?
In such scenario, how can we correct set `isRemoteVsan` field?

Reply:
As per the offline discussion since cross zonal datastores are not supported, I've updated this func to iterate over all the clusters(which will be only one in case of non-stretched) and return false if any cluster has matching UUID.
Otherwise, we can safely consider that the datastore is remote.